activesupport/lib/active_support/deprecation/proxy_wrappers.rb

```diff
@@ -53,15 +53,15 @@ def warn(callstack, called, args)
         end
     end
 
-    # DeprecatedInstanceVariableProxy transforms an instance variable into a
-    # deprecated one. It takes an instance of a class, a method on that class
-    # and an instance variable. It optionally takes a deprecator as the last
-    # argument. The deprecator defaults to +ActiveSupport::Deprecator+ if none
-    # is specified.
+    # DeprecatedInstanceVariableProxy transforms an instance variable into a deprecated one. It takes an instance of a
+    # class, a method on that class, an instance variable, and a deprecator as the last argument.
+    #
+    # Trying to use the deprecated instance variable will result in a deprecation warning, pointing to the method as a
+    # replacement. This allows to change an API based on an instance variable access to a method call.
```

Comment:
```suggestion
    # Trying to use the deprecated instance variable will result in a deprecation warning, pointing to the method as a
    # replacement.
```

I think the second sentence here is a little repetitive of the previous sentence.